src/components/MintModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import { Mesh } from "../contracts/Mesh";
+import { Button, Modal } from "../components";
+
+type Props = {
+  isModalOpen: boolean;
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  contract: Mesh;
+};
+
+const MintModal: React.FC<Props> = (props) => {
+  const { isModalOpen, setIsModalOpen, contract } = props;
+
+  const [error, setError] = useState<any>();
+  const [loading, setLoading] = useState(false);
+  const [address, setAddress] = useState("");
+  const [timestamp, setTimestamp] = useState<number>();
+  const [transaction, setTransaction] = useState("");
+
+  const mint = async () => {
+    const matchingAddress = address.match(/^0x[a-fA-F0-9]{40}$/);
```

Comment:
You can also use `ethers.utils.isAddress( address ) ⇒ boolean` to check for valid addresses

Reply:
Oh nice! Will add